src/coord/src/coord.rs

```diff
@@ -1251,7 +1265,10 @@ where
 
             let (project, filter) = Self::plan_peek(source.as_mut());
 
-            let (fast_path, index_id) = if let RelationExpr::Get {
+            // If an `as_of` argument is provided, always build a new transient dataflow.
+            let (fast_path, index_id) = if as_of.is_some() {
+                (false, self.catalog.allocate_id()?)
```

Comment:
I'm afraid I don't follow the reasoning for this. If the timestamp is currently valid, why can't we avoid the dataflow in the same way that the other case does?

Reply:
fixed this!